### PR TITLE
Sunshine New Users refactor + variable snooze length

### PR DESCRIPTION
**Pull request by Raemon**
This is a WIP after working with Vlad (a programming tutor). I had previously attempted something similar to this a year ago and found it hard to wrap my head around all the cases the sunshine New Users review system currently handles.

While working on it, I realized I really should have a chat with other moderators and EA forum staff about what we actually want out of the moderation system. I'm putting the current version up to illustrate some things worth talking about. (I haven't tested the current version and don't assume it's working yet)

The actual new feature here is variable snooze length. It adds a "nextReviewContentCount" to the user object, and snoozing sets the nextReviewContentCount to "postCount + commentCount + N" (with a plan for buttons "snooze +1" and "snooze +10"). Getting this to work properly requires some migrations, and probably thinking through edge cases about re-drafting that I haven't thought about yet.

This PR also aims to fix the "re-drafting a post doesn't properly remove you from the sunshine queue, if you have no other outstanding posts."

In the process of looking at that, Vlad and I noted that the code for "what actually results in users ending up in the New Users queue" is distributed among lots of callbacks, in a way that makes it hard to keep track of. We tried moving all the cases to a single function, and then just have each callback trigger that function.

There's some philosophical question here about whether that function should require any information other than the user object. (It's convenient if so).

### CODE REVIEW

File: packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx [hunk 1]
```diff
@@ -140,6 +141,22 @@ const styles = (theme: ThemeType): JssStyles => ({
...
+  commentCount?: number
+}
+
+export function getCurrentContentCount(user: UserContentCountPartial) {
+  const postCount = user.postCount || 0
+  const commentCount = user.commentCount || 0
+  return postCount + commentCount;
+}
+
+export function getNewSnoozeUntilContentCount(user: UserContentCountPartial, contentCount: any) {
```
Comment (b0b3rt) [inline packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx:156]: `contentCount: number`?

File: packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx [hunk 2]
```diff
@@ -140,6 +141,22 @@ const styles = (theme: ThemeType): JssStyles => ({
...
 })
+
+export interface UserContentCountPartial {
+  postCount?: number,
+  commentCount?: number
+}
+
+export function getCurrentContentCount(user: UserContentCountPartial) {
+  const postCount = user.postCount || 0
+  const commentCount = user.commentCount || 0
```
Comment (b0b3rt) [inline packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx:152]: nit: `??` for both of these

File: packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx (hunk 1 above)
Comment (Raemon) [reply packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx:156]: nah

File: packages/lesswrong/server/callbacks/userCallbacks.tsx [hunk 3]
```diff
@@ -91,16 +92,7 @@ getCollectionHooks("Users").editAsync.add(async function approveUnreviewedSubmis
...
-  const addedBio = !oldUser.biography?.html && newUser.biography?.html && forumTypeSetting.get() === 'EAForum'
-  
-  // on the EA Forum, we are reviewing all unreviewed users who add a profile photo
-  const addedProfilePhoto = !oldUser.profileImageId && newUser.profileImageId && forumTypeSetting.get() === 'EAForum'
-
-  // if the user has a mapLocation and they have not been reviewed, mark them for review
-  if ((addedBio || addedProfilePhoto || newUser.mapLocation) && !newUser.reviewedByUserId && !newUser.needsReview) {
-    void Users.rawUpdateOne({_id: newUser._id}, {$set: {needsReview: true}})
-  }
+  return triggerReviewIfNeeded(newUser._id)
```
Comment (Raemon) [inline packages/lesswrong/server/callbacks/userCallbacks.tsx:95]: I'm not 100% sure it's kosher to return something other than a user object here

File: packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx (hunk 1 above)
Comment (b0b3rt) [reply packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx]: ?

Comment (Raemon) [reply packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx]: I... do not remember why I thought it was appropriate to reply "nah" (it's not even a particularly good joke), but apparently I already updated it to use "number." Wuckles.

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**